### PR TITLE
RTL Conversion for NxCheckbox - RSC-1237

### PR DESCRIPTION
**Pull request by cindyko226**
https://issues.sonatype.org/browse/RSC-1237

### CODE REVIEW

File: lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx [hunk 1]
```diff
@@ -4,167 +4,92 @@
...
+        isChecked: false
+      },
+      quickRender = rtlRender(NxCheckbox, minimalProps),
+      renderEl = rtlRenderElement(NxCheckbox, minimalProps);
+
+  it('renders a <label> containing a checkbox role <input>', function() {
+    const component = renderEl()!,
+        input = within(component).getByRole('checkbox');
+
+    expect(component.tagName).toBe('LABEL');
```
Comment (rpokorny) [inline lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx:25]: I tend to think the label element is an implementation detail.  We do want to check that it gets connected as the accessible name of the input though.

File: lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx [hunk 2]
```diff
@@ -4,167 +4,92 @@
...
+    const el = renderEl({ id: 'foo', lang: 'en-US' });
 
-  it('adds the nx-radio-checkbox--disabled class if disabled is set', function() {
-    expect(getShallowComponent()).not.toHaveClassName('nx-radio-checkbox--disabled');
-    expect(getShallowComponent({ disabled: true })).toHaveClassName('nx-radio-checkbox--disabled');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en-US');
   });
 
-  it('adds the tm-checked class if isChecked is true, and the tm-unchecked class if it is false', function() {
```
Comment (rpokorny) [inline lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx:68]: This test should be kept/updated.  These classes are for external user analytics needs

File: lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx (hunk 1 above)
Comment (cindyko226) [reply lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx]: Updated here
a42ddb52a04ffc150ea2260c27f66d569f93b25c
b6036817b68162b098ddabae1831cb0171dd3cbe

File: lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx (hunk 2 above)
Comment (cindyko226) [reply lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx:68]: Updated here 784425a7d80cafc654eadcb4bbf42a4371e06189, thanks.

File: lib/src/components/NxCheckbox/stateful/__tests__/NxStatefulCheckbox.test.tsx [hunk 3]
```diff
@@ -4,57 +4,112 @@
...
+
+  it('updates checked state when the checkbox is toggled', async function() {
+    const user = userEvent.setup(),
+        input = quickRender().getByRole('checkbox');
+
+    expect(input).not.toBeChecked();
+
+    await user.click(input);
+
+    expect(input).toBeChecked();
```
Comment (rpokorny) [inline lib/src/components/NxCheckbox/stateful/__tests__/NxStatefulCheckbox.test.tsx:101]: Might as well make sure it toggles back too

**Review by rpokorny — APPROVED** [review]

File: lib/src/components/NxCheckbox/stateful/__tests__/NxStatefulCheckbox.test.tsx (hunk 3 above)
Comment (cindyko226) [reply lib/src/components/NxCheckbox/stateful/__tests__/NxStatefulCheckbox.test.tsx:101]: Updated here db2b7763ab07a078d3b9cedf9335868acdd77ccf. Thanks.

**Pull request merged**